homeassistant/components/rfxtrx/__init__.py

```diff
@@ -191,7 +191,11 @@ async def async_setup_entry(hass, entry: config_entries.ConfigEntry):
 
     hass.data[DOMAIN][DATA_CLEANUP_CALLBACKS] = []
 
-    await async_setup_internal(hass, entry)
+    try:
+        await async_setup_internal(hass, entry)
+    except ConfigEntryNotReady:
+        # Library currently doesn't support reload
+        return False        
```

Comment:
Well it's just the timeout case that doesn't support reload really. So I would probably just not throw the not ready exception instead. Just some generic exception.

Reply:
I agree. We shouldn't reuse the HA exceptions for integration flow control.

Reply:
I thought I'd keep the infrastructure as is to remove those lines once library supports it.

Alternatively we could remove the catch in `async_setup_internal` and catch the asyncio.TimeoutError here.

Reply:
Just raise Exception("Unhandable timeout in library") where it now raises the not ready exception and let that propagate out. No need to get fancy here. It's a error situation.